src/main/java/com/iexec/core/result/ResultController.java

```diff
@@ -106,21 +106,27 @@ public ResultController(ResultService resultService,
     @CrossOrigin
     @GetMapping(value = "/results/{chainTaskId}", produces = "application/zip")
     public ResponseEntity<byte[]> getResult(@PathVariable("chainTaskId") String chainTaskId,
-                                            @RequestHeader("Authorization") String token,
+                                            @RequestHeader(name = "Authorization", required = false) String token,
                                             @RequestParam(name = "chainId") Integer chainId) throws IOException {
         Authorization auth = authorizationService.getAuthorizationFromToken(token);
 
-        if (!(authorizationService.isAuthorizationValid(auth) &&
-                resultService.canGetResult(chainId, chainTaskId, auth.getWalletAddress()))) {
-            return new ResponseEntity(HttpStatus.UNAUTHORIZED);
-        }
+        boolean isPublicResult = resultService.isPublicResult(chainTaskId, chainId);
+        boolean isOwnerOfResultAndAuthorized = auth != null
```

Comment:
What about _isAuthorizedOwnerOfResult_ ?

Reply:
As you wish!